src/CheckoutSdk/Marketplace/MarketplaceClient.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Checkout.Common;
+using Checkout.Files;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Checkout.Marketplace
+{
+    public class MarketplaceClient : AbstractClient, IMarketplaceClient
+    {
+        public const string MarketplacePath = "marketplace";
+        public const string EntitiesPath = "entities";
+        public const string InstrumentPath = "instruments";
+        public const string FilesPath = "base:files";
+
+        private IFilesClient FilesClient { get; set; }
+
+        public MarketplaceClient(
+            IApiClient apiClient, 
+            ICheckoutConfiguration configuration,
+            IFilesClient filesClient)
+            : base(apiClient, configuration, SdkAuthorizationType.OAuth)
+        {
+            this.FilesClient = filesClient;
+        }
+
+        public async Task<OnboardEntityResponse> CreateEntity(OnboardEntityRequest entityRequest)
+        {
+            CheckoutUtils.ValidateParams("entityRequest", entityRequest);
+            return await ApiClient.Post<OnboardEntityResponse>(BuildPath(MarketplacePath, EntitiesPath), SdkAuthorization(), entityRequest);
+        }
+
+        public async Task<OnboardEntityDetailsResponse> GetEntity(string entityId)
+        {
+            CheckoutUtils.ValidateParams("entityId", entityId);
+            return await ApiClient.Get<OnboardEntityDetailsResponse>(BuildPath(MarketplacePath, EntitiesPath, entityId), SdkAuthorization());
+        }
+
+        public async Task<OnboardEntityResponse> UpdateEntity(OnboardEntityRequest entityRequest, string entityId)
+        {
+            CheckoutUtils.ValidateParams("entityRequest", entityRequest, "entityId", entityId);
+            return await ApiClient.Put<OnboardEntityResponse>(BuildPath(MarketplacePath, EntitiesPath, entityId), SdkAuthorization(), entityRequest);
+        }
+
+        public async Task CreatePaymentInstrument(MarketplacePaymentInstrument marketplacePaymentInstrument, string entityId)
+        {
+            CheckoutUtils.ValidateParams("marketplacePaymentInstrument", marketplacePaymentInstrument, "entityId", entityId);
+            await ApiClient.Post<bool>(BuildPath(MarketplacePath, EntitiesPath, entityId, InstrumentPath), SdkAuthorization(), marketplacePaymentInstrument);
```

Comment:
Should we return `bool` is this case? What was the solution found for this use cases? @Armando-Ibarra

Reply:
I think that the actual response is an `Id` said that `IdResponse` should be enough.